src/applications/mhv-supply-reordering/actions/index.js

```diff
@@ -0,0 +1,45 @@
+import { apiRequest } from 'platform/utilities/api';
+import head from 'lodash/head';
+import { MDOT_API_STATES, MDOT_API_URL } from '../constants';
+
+const handleError = (statusCode, errorCode) => ({
+  type: MDOT_API_STATES.ERROR,
+  statusCode,
+  errorCode,
+});
+
+const handleSuccess = data => ({
+  type: MDOT_API_STATES.SUCCESS,
+  statusCode: '200',
+  data,
+});
+
+const initiateApiCall = () => ({
+  type: MDOT_API_STATES.PENDING,
+});
+
+/**
+ * Fetch MDOT data. The data or errors are stored as part of the dispatch payload.
+ * @returns a dispatch
+ */
+// TODO: Report errors
+export const fetchMdotData = () => async dispatch => {
+  dispatch(initiateApiCall());
+  apiRequest(MDOT_API_URL)
+    .then(body => {
+      if (body.errors) {
+        const mdotError = head(body.errors);
+        return dispatch(handleError(mdotError.status, mdotError.code));
+      }
+      return dispatch(handleSuccess(body.formData));
+    })
+    .catch(error => {
+      if (error.errors) {
+        const mdotError = head(error.errors);
```

Comment:
I don't think you need lodash for this. I think `const [mdotError = null] = body?.errors` or `const mdotError = body?.errors[0]` (if you're okay with the value being `null` or `undefined`) would get you the possible first element in `error.errors`

Reply:
Makes sense. Will update this in the next PR.